src/common/IterationLookup.ts

```diff
@@ -0,0 +1,280 @@
+        let endDate = DateTime.fromFormat(
+          `${endMonth} ${match[4]} ${year}`,
+          "LLL d y",
+          { locale: "en-US" }
+        );
+        if (endDate.month < lastMonth) {
+          year += 1;
+        }
+        // If the end date is a Monday, set it to the previous Sunday.
+        if (endDate.weekday === 1) {
+          endDate = endDate.minus({ days: 1 });
+        }
+        // Otherwise, set it to the next Sunday.
+        endDate = endDate.set({ year }).endOf("week");
+        lastDate = endDate;
+        lastMonth = lastDate.month;
+        const endDateTime = endDate.startOf("day").toISO();
+        const weeks = Math.ceil(endDate.diff(startDate, "days").days / 7);
+        if (startDateTime && endDateTime && weeks) {
+          this.byVersionString[iteration] = {
+            startDate: startDateTime,
+            weeks,
+            endDate: endDateTime,
+          };
+          this.orderedVersionStrings.push(iteration);
+          const start = DateTime.fromISO(startDateTime);
+          for (let i = 0; i < weeks; i++) {
+            const monday = start.plus({ weeks: i });
+            this.byDate[monday.toFormat("yyyy-MM-dd")] = iteration;
+          }
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Convert a month string to a 3-letter abbreviation. Some month strings from
+ * Bugzilla iterations are full month strings or 4-letter abbreviations like
+ * "Sept", so we need to convert them to abbreviations that luxon can parse.
+ * @param {string} month
+ * @returns {string}
+ * @example "September" => "Sep"
+ * @example "Sept" => "Sep"
+ * @example "AUG" => "Aug"
+ */
+function normalizeMonthString(month: string): string {
+  return month
+    .slice(0, 3)
+    .toLowerCase()
+    .replace(/^./, c => c.toUpperCase());
+}
+
+/**
+ * A class that provides utilities for looking up iterations by date or version.
+ * While IterationLookup is passed through network requests, Iterations is
+ * constructed on the client side.
+ */
+export class Iterations implements IterationLookup {
+  byDate: { [date: string]: string };
+  byVersionString: {
+    [versionString: string]: {
+      startDate: string;
+      endDate: string;
+      weeks: number;
+    };
+  };
+  orderedVersionStrings: string[];
+  // Utility to get the number of work days between two dates
+  getWorkDays: (startDate: Date | string, endDate: Date | string) => number;
+  // Utility to get the Monday before a given date
+  getMondayBefore: (date: DateTime) => DateTime;
+
+  constructor(iterationLookup: IterationLookup) {
+    this.byVersionString = iterationLookup.byVersionString;
+    this.orderedVersionStrings = iterationLookup.orderedVersionStrings;
+    this.byDate = iterationLookup.byDate;
+    this.getWorkDays = Iterations.getWorkDays.bind(this);
+    this.getMondayBefore = Iterations.getMondayBefore.bind(this);
+    this.getIteration = this.getIteration.bind(this);
+    this.getAdjacentIteration = this.getAdjacentIteration.bind(this);
```

Comment:
Do we need to explicitly bind these? We're not passing the methods as raw callbacks anywhere so this seems unnecessary.